### PR TITLE
[Dominion] Profile Releases Section

### PR DESCRIPTION
**Pull request by karimkawambwa**
## Description

Release section.

Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screen Shot 2019-03-13 at 2 35 59 PM
Screen Shot 2019-03-13 at 2 35 41 PM

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by karimkawambwa** [comment]
I have gone through the code but I am not sure what is the variable that holds the available releases. I can see releases setup in settings and and `profile_detail.html` has profileData variable has 

```
"primary_releases": {
		"active": {
			"citation": "Census, 2016",
			"name": "Census",
			"year": "2016"
		},
		"other": []
	}
```

Should i just show the active in the dropdown?
cc @KhadijaMahanga @kilemensi 




Full Json Below:
```{
	"demographics": {
		"total_population": {
			"index": {
				"this": null
			},
			"name": "People",
			"error_ratio": {
				"this": null
			},
			"values": {
				"this": 0
			},
			"numerators": {
				"this": 0,
				"district": 0,
				"province": 0,
				"country": 0
			},
			"error": {
				"this": 0
			},
			"numerator_errors": {
				"this": 0,
				"district": 0,
				"province": 0,
				"country": 0
			}
		},
		"is_missing": true
	},
	"primary_releases": {
		"active": {
			"citation": "Census, 2016",
			"name": "Census",
			"year": "2016"
		},
		"other": []
	},
	"geography": {
		"this": {
			"geo_code": "ZA",
			"child_level": "level1",
			"version": "2009",
			"short_name": "South Africa",
			"name": "South Africa",
			"full_name": "South Africa",
			"parent_geoid": "continent-AFR",
			"full_geoid": "country-ZA",
			"square_kms": 1221037.0,
			"geo_level": "country"
		},
		"parents": {
			"continent": {
				"geo_code": "AFR",
				"child_level": "country",
				"version": "2009",
				"short_name": "Africa",
				"name": "Africa",
				"full_name": "Africa",
				"parent_geoid": null,
				"full_geoid": "continent-AFR",
				"square_kms": null,
				"geo_level": "continent"
			}
		},
		"comparatives": []
	}
}```

**Comment by kilemensi** [comment]
Yes @karimkawambwa, `primary_releases` is what you need. `active` is the one you show as selected/current and the dropdown should contain everything (`other` + `active`)

**Comment by karimkawambwa** [comment]
The only thing left is getting metadata for citations which i am not sure where I can get it. Or should i just change the url.

```
profileData={
    "education": {
        "educational_attainment": {
            "percent_further_edu": {
                "values": {
                    "this": 32.0,
                    "country": 43.37
                },
                "name": "Completed Matric or higher",
                "index": {
                    "this": 100.0,
                    "country": 74.0
                },
                "error_ratio": {
                    "this": null,
                    "country": null
                },
                "error": {
                    "this": 0,
                    "country": 0
                },
                "numerator_errors": {
                    "this": 0,
                    "country": 0
                },
                "numerators": {
                    "this": 1199232,
                    "country": 14644033
                }
            },
            "percent_general_edu": {
                "values": {
                    "this": 64.66,
                    "country": 71.77
                },
                "name": "Completed Grade 9 or higher",
                "index": {
                    "this": 100.0,
                    "country": 90.0
                },
                "error_ratio": {
                    "this": null,
                    "country": null
                },
                "error": {
                    "this": 0,
                    "country": 0
                },
                "numerator_errors": {
                    "this": 0,
                    "country": 0
                },
                "numerators": {
                    "this": 2423458,
                    "country": 24232255
                }
            },
            "metadata": {
                "table_id": "HIGHESTEDUCATIONALLEVEL20",
                "release": "Community Survey 2016",
                "citation": "Statistics South Africa (2016) South African Community Survey 2016. Indicators derived from the full population Community Survey.",
                "universe": "Individuals 20 and older",
                "year": "2016"
            }
        },
```

cc @kilemensi @KhadijaMahanga

**Review by KhadijaMahanga — CHANGES_REQUESTED** [review]
1. Dropdown position keeps changing. 

Here its on the right
Screenshot from 2019-03-14 11-49-30

Here it covers the release dropdown
Screenshot from 2019-03-14 11-49-57

2. Active release is `Census`. But the template is still showing the Community Survey (still static)

3. The release dropdown icon is not changing on clicked (ArrowUp/ArrowDown)

**Comment by karimkawambwa** [comment]
@kilemensi I am not sure what this means `Active release is Census. But the template is still showing the Community Survey (still static)`. I you mean the metadata, the citations then i actually don't now how to get these or where they are coming from. I will check with @KhadijaMahanga's  profile header.

**Comment by KhadijaMahanga** [comment]
> @kilemensi I am not sure what this means `Active release is Census. But the template is still showing the Community Survey (still static)`. I you mean the metadata, the citations then i actually don't now how to get these or where they are coming from. I will check with @KhadijaMahanga's profile header.

@karimkawambwa I meant this variable
```
"primary_releases": {
		"active": {
			"citation": "Census, 2016",
			"name": "Census",
			"year": "2016"
		},
		"other": []
	}
```

Our active release is Census. So the release section should be describing census not community survey.

**Comment by karimkawambwa** [comment]
> Our active release is Census. So the release section should be describing census not community survey.

~~@KhadijaMahanga thats why am not sure where to get that metadata, I am combing through code.
In wazimap i see this but i don't see it in hurumap~~

 ``` ,
            "metadata": {
                "table_id": "HIGHESTEDUCATIONALLEVEL20",
                "release": "Community Survey 2016",
                "citation": "Statistics South Africa (2016) South African Community Survey 2016. Indicators derived from the full population Community Survey.",
                "universe": "Individuals 20 and older",
                "year": "2016"
            }
```  

**Ignore this comment.**
looks like these citations are static.
I will be changing the title.

**Comment by karimkawambwa** [comment]
@KhadijaMahanga 👀

File: dominion/dominion_ui/src/components/ProfileReleases/index.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    }));
+  }
+
+  render() {
+    const { classes } = this.props;
+    const { menuOpen } = this.state;
+    const datasetRealeases = window.dataset_releases;
+    const activeDataset = window.active_dataset;
+
+    const Atag = ({ link }) => (
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/ProfileReleases/index.js]: We have `A` component for links (it uses theme settings, etc.): https://github.com/CodeForAfrica/HURUmap-apps/blob/fe08690b233717f0fbc2f49d3888f0005db41bf8/dominion/dominion_ui/src/components/A.js#L9

File: dominion/dominion_ui/src/components/ProfileReleases/index.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    fontFamily: 'Montserrat',
+    fontSize: '14px',
+    fontWeight: 'bold',
+    fontStyle: 'normal',
+    fontStretch: 'normal',
+    lineHeight: 1.43,
+    letterSpacing: 'normal',
+    color: '#2c2c2a',
+    padding: '8px 0'
+  },
```
Comment (kilemensi) [inline dominion/dominion_ui/src/components/ProfileReleases/index.js:54]: Again, why are you not using theme typography for `font` stuff?

File: dominion/templates/profile/profile_detail.html [hunk 3]
```diff
@@ -10,9 +10,18 @@
...
+
+
+{% block body_javascript_extra %}
+    {{ block.super }}
+    <script>
+        var dataset_releases = profileData["primary_releases"]["other"];
+        dataset_releases.push(profileData["primary_releases"]["active"]);
+        var current_dataset = rofileData["primary_releases"]["active"];
+    </script>
+{% endblock %}
```
Comment (kilemensi) [inline dominion/templates/profile/profile_detail.html:26]: New line

Comment (kilemensi) [reply dominion/templates/profile/profile_detail.html:26]:
```javascript
        var current_dataset_release = profileData["primary_releases"]["active"];
        var dataset_releases = profileData["primary_releases"]["other"].concat(current_dataset_release);
```

This way, we don't mutate `profileData["primary_releases"]["other"]`.

File: dominion/dominion_ui/src/components/ProfileReleases/index.js (hunk 2 above)
Comment (karimkawambwa) [reply dominion/dominion_ui/src/components/ProfileReleases/index.js:54]: From the talk, we shall walk through to move thing to their appropriate theme.

**Pull request merged**